### PR TITLE
Fix Release Date

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
- Change release date from Instant to LocalDate in order to expect a more possible use case which is "YYYY-MM-DD" instead of expecting the time to be provided
- Add custom exception with a proper error message which indicates the expected format
- Add missing column annotation in the persistence layer
- Add missing configuration properties
- Fix some typos and improve method names

### CODE REVIEW

File: commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SubExt.kt [hunk 1]
```diff
@@ -22,15 +19,16 @@ var Submission.attachTo: String?
...
-    }
-
+var Submission.releaseDate: String?
+//    get() = get<String?>(SubFields.RELEASE_DATE)?.let {
+//        try {
+//            LocalDate.parse(it)
+//        } catch (exception: DateTimeParseException) {
+//            throw InvalidDateFormatException(it)
+//        }
+//    }
```
Comment (Juan-EBI) [inline commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SubExt.kt]: :fearful:

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/TsvDeserializer.kt [hunk 2]
```diff
@@ -47,6 +47,7 @@ class TsvDeserializer(private val chunkProcessor: ChunkProcessor = ChunkProcesso
 
     private fun chunkerize(pagetab: String) =
         pagetab.split(TSV_CHUNK_BREAK)
+            .map { it.removeSuffix("\r") }
```
Comment (Juan-EBI) [inline commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/TsvDeserializer.kt]: can you explain this one please

File: submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/processors/TimesProcessor.kt [hunk 3]
```diff
@@ -13,14 +16,25 @@ import java.time.ZoneOffset.UTC
...
+        var releaseDate = OffsetDateTime.now()
+        submission.releaseDate?.let {
+            try {
+                releaseDate = LocalDate.parse(it).atStartOfDay().atOffset(ZoneOffset.UTC)
+            } catch (exception: DateTimeParseException) {
+                throw InvalidDateFormatException(it)
+            }
+        }
+
+        return releaseDate
```
Comment (Juan-EBI) [inline submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/processors/TimesProcessor.kt]: what about
`submission.releaseDate?.let { parseDate(it) } ?: now`

and
```
fun parseDate(date:String): OffsetDateTime {
    return  Try.just(LocalDate.parse(it)
                    .atStartOfDay()
                    .atOffset(UTC))
                    .fold({it}, { throw InvalidDateFormatException(date) })
}
```

File: submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/exceptions/InvalidDateFormatException.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package ac.uk.ebi.biostd.submission.exceptions
+
+class InvalidDateFormatException(
+    date: String
+) : RuntimeException("Invalid date format provided for date $date. Expected format is YYYY-MM-DD")
```
Comment (Juan-EBI) [inline submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/exceptions/InvalidDateFormatException.kt]: it looks confusing for me, maybe

`Provided date `$date` could not be parsed. Expected format is YYYY-MM-DD`

File: commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SubExt.kt (hunk 1 above)
Comment (jhoanmanuelms) [reply commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SubExt.kt]: 🤦‍♂

File: submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/processors/TimesProcessor.kt (hunk 3 above)
Comment (jhoanmanuelms) [reply submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/processors/TimesProcessor.kt]: 👍

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/TsvDeserializer.kt (hunk 2 above)
Comment (jhoanmanuelms) [reply commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/TsvDeserializer.kt]: This is related to MacOS end of line encoding. We just need to remove it if present.

**Pull request merged**